tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc

```diff
@@ -9111,6 +9111,433 @@ TEST_P(OpConverter_FP32_FP16_Test, ConvertPad) {
+        {-1,  2, 1, -4,  1, -6,  -7,  2, 1,   -10, -11, 2,
+         -13, 2, 1, -16, 1, -18, -19, 2, -21, 2,   1,   -24},
+        {-1, 2,  -1, 4,  -1, 6,  -1, 8,  -1, 10, -1, 12,
+         -1, 14, -1, 16, -1, 18, -1, 20, -1, 22, -1, 24},
+        {-1,  1, 2, -4,  3, -6,  -7,  1, 2,   -10, -11, 3,
+         -13, 1, 2, -16, 3, -18, -19, 1, -21, 2,   3,   -24},
+        {-1, -1, 3,  4,  5,  6,  -1, -1, 9,  10, 11, 12,
+         -1, -1, 15, 16, 17, 18, -1, -1, 21, 22, 23, 24},
+        {-1,  1, 1, -4,  1, -6,  -7,  2, 2,   -10, -11, 2,
+         -13, 3, 3, -16, 3, -18, -19, 4, -21, 4,   4,   -24},
+        {-1, -1, -1, -1, -1, -1, 7,  8,  9,  10, 11, 12,
+         13, 14, 15, 16, 17, 18, -4, -4, -4, -4, -4, -4},
+        {-1,  1, 1, -4,  1, -6,  -7,  1, 1,   -10, -11, 1,
+         -13, 1, 1, -16, 1, -18, -19, 1, -21, 1,   1,   -24},
+        {-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
+         -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1},
+        {-1, 2,  -1, 4,  -1, 6,  1,  -4, 3,  -4, 5,  -4,
+         1,  -6, 3,  -6, 5,  -6, -7, 2,  -7, 4,  -7, 6},
+        {-1, 2, 1, -4, 1, -6, -1, 4, 3, -4, 3, -6,
+         -1, 6, 5, -4, 5, -6, -1, 8, 7, -4, 7, -6}};
+
+    const auto exp_dims = dims + 3;
+    const int kMax2 = 2;  // number of permutations on 2 elements
+    iMax = sizeof(feasible_dims_2) / sizeof(feasible_dims_2[0]);
+    assert(kMax2 * iMax / 3 ==
+           sizeof(expected_val_2) / sizeof(expected_val_2[0]));
+    // Broadcast shapes defined for `cond` OR for `then` and `else`.
+    // Loop for all pairs of nvinfer1::Dims from feasible_dims_2.
+    for (int i = 0; i < iMax; i += 2) {
+      // Loop for all permutations on 2 elements.
+      for (int k = 0; k < kMax2; k++) {
+        // Constructing dims for tensors 'cond' and 'then'.
+        // NOTE: dims('else') will be the same as  dims('then').
+        for (int j = 0; j < 2; j++)
+          set_dimension(feasible_dims_2 + i + (j + k) % 2, dims[j]);
+
+        std::vector<float>* expect = expected_val_2 + i + k;
+        // Loop where the tensor shapes for 'cond' and 'then' are swapping.
+        for (int n = 0; n < 2; n++) {
+          assign_values(par_dims, par_value, data_cond, exp_dims, true, expect);
+          set_parameters(config, par_name, par_value, par_type, par_dims,
+                         data_cond);
+          TestOpConverter("my_select", node, *exp_dims, status, status,
+                          ElementsAreArray(expected_output));
+          // Swapping dims for 'cond' and 'then' tensors.
+          const auto tmp = par_dims[0];
+          par_dims[0] = par_dims[1];
+          par_dims[1] = tmp;
+          expect += iMax;
+        }
+      }
+    }
+
+    // Tests for pairwise different dims('cond'), dims('then'), dims('else').
+    nvinfer1::Dims feasible_dims_3[] = {
+        {2, {3, 2}},    {2, {3, 1}},    {2, {1, 1}},
+        {3, {2, 2, 1}}, {3, {2, 1, 2}}, {3, {1, 2, 2}},
+        {3, {2, 1, 1}}, {3, {2, 1, 2}}, {3, {1, 2, 2}},
```

Comment:
Please add  {3, {2, 1, 1}}, {3, {1, 1, 2}}, {3, {1, 2, 1}} to the test.

Reply:
Done.